### ISSUE TITLE
Allow passing string literals to TypeVarTuple callbacks

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#6635]
**Is your feature request related to a problem? Please describe.**
It would be awesome if the following would work with pyright. It does with mypy 1.7.1.
```py
from typing import Literal, TypeVarTuple, Callable

Ts = TypeVarTuple("Ts")

def func(cb: Callable[[*Ts], None], *args: *Ts) -> None: ...
def g(a: Literal["day", "hour"]) -> None: ...

func(g, "day")  # pyright error
```
```
Argument of type "*tuple[str]" cannot be assigned to parameter "args" of type "*Ts@func" in function "func"
```

Even better, if this would work as well. However, it doesn't work even for mypy at the moment.
```py
x: bool
func(g, "day" if x else "hour")
```

--
Both patterns work fine if the string literals are passed to `g` directly.
```py
g("day")
g("day" if x else "hour")
```

**Comment by cdce8p** [comment]
Another extension would be something like this
```py
def h(a: set[Literal["sum", "min"]]) -> None: ...
h({"sum"})
func(h, {"sum"})  # error
```

**Comment by erictraut** [comment]
The reason this doesn't work is that pyright doesn't apply bidirectional type inference when evaluating the type of the `"day"` argument. By normal inference rules, the type of this expression is `str`.

Adding support for bidirectional type inference for this particular case is theoretically possible, but it adds significant complexity to a code path that is already very complicated. I see that mypy also doesn't support general bidirectional type inference for this case.

```python
from typing import TypeVarTuple, Callable, TypedDict

Ts = TypeVarTuple("Ts")

def func(cb: Callable[[*Ts], None], *args: *Ts) -> None: ...

class User(TypedDict):
    name: str

def g(a: User) -> None: ...

func(g, {"name": "Eric"}) # Both pyright and mypy generate an error here
```

I realize that you are not asking for general bidirectional type inference support in this case. I mention that because it's the "correct" way to implement what you're asking for.

There is a relatively simple fix that addresses your specific ask that enables support for literals. I've taken this approach which addresses your ask and provides parity with mypy's capabilities. This will be included in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.339, which I just published. It will also be included in a future release of pylance.
